src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1280,27 +1287,93 @@ TIP: If the executing Node has no Docker installed, the task will fail. A <<_sel
+cmd.add("run")
+cmd.add("--rm")
+cmd.add("--env")
+cmd.add("HOME=/tmp")
+
+String osName = System.getProperty("os.name");
+println "Operating system : " + osName;
+OperatingSystem operatingSystem = OperatingSystem.resolveOrError(osName);
+OperatingSystemFamily family = operatingSystem.getFamily();
+
+switch (family) {
+    case OperatingSystemFamily.WINDOWS:
+    	isWindows = true;
+    	break;
+    default:
+        isWindows = false;
+}
+forkEnvironment.setDockerWindowsToLinux(isWindows)
+
+// Prepare ProActive home volume
 paHomeHost = variables.get("PA_SCHEDULER_HOME")
-paHomeContainer = variables.get("PA_SCHEDULER_HOME")
-proActiveHomeVolume = '-v '+paHomeHost +':'+paHomeContainer+' '
-# Prepare working directory (For Dataspaces and serialized task file)
+paHomeContainer = (isWindows ? forkEnvironment.convertToLinuxPath(variables.get("PA_SCHEDULER_HOME")) : variables.get("PA_SCHEDULER_HOME"))
+cmd.add("-v")
+cmd.add(paHomeHost + ":" + paHomeContainer)
+// Prepare working directory (For Dataspaces and serialized task file)
 workspaceHost = localspace
-workspaceContainer = localspace
-workspaceVolume = '-v '+localspace +':'+localspace+' '
-# Prepare container working directory
-containerWorkingDirectory = '-w '+workspaceContainer+' '
-# Save pre execution command into magic variable 'preJavaHomeCmd', which is picked up by the node
-preJavaHomeCmd = dockerRunCommand + dockerParameters + proActiveHomeVolume + workspaceVolume + containerWorkingDirectory + containerName
+workspaceContainer = (isWindows ? forkEnvironment.convertToLinuxPath(workspaceHost) : workspaceHost)
+cmd.add("-v")
+cmd.add(workspaceHost + ":" + workspaceContainer)
+
+cachespaceHost = cachespace
+cachespaceContainer = (isWindows ? forkEnvironment.convertToLinuxPath(cachespaceHost) : cachespaceHost)
+cmd.add("-v")
+cmd.add(cachespaceHost + ":" + cachespaceContainer)
+
+if (!isWindows) {
+     // when not on windows, mount and use the current JRE
+    currentJavaHome = System.getProperty("java.home")
+	forkEnvironment.setJavaHome(currentJavaHome)
+    cmd.add("-v")
+    cmd.add(currentJavaHome + ":" + currentJavaHome)
+}
+
+// Prepare container working directory
+cmd.add("-w")
+cmd.add(workspaceContainer)
+
+if (isWindows) {
+    // linux on windows does not allow sharing identities (such as AD identities)
+} else {
+    sigar = new org.hyperic.sigar.Sigar()
+    try {
+        pid = sigar.getPid()
+        creds = sigar.getProcCred(pid)
+        uid = creds.getUid()
+        gid = creds.getGid()
+        cmd.add("--user=" + uid + ":" + gid)
+    } catch (Exception e) {
+        println "Cannot retrieve user or group id : " + e.getMessage()
+    } finally {
+        sigar.close()
+    }
+}
+cmd.add(containerName)
+
+forkEnvironment.setPreJavaCommand(cmd)
+
+// Show the generated command
+println "DOCKER COMMAND : " + forkEnvironment.getPreJavaCommand()
 ----
 This script exports a variable called `preJavaHomeCmd`, which is picked up by the Node executing the Task.
```

Comment:
preJavaHomeCmd is not used any more(deprecated). We use now forkEnvironment.setPreJavaCommand with a list of command + arguments